cluster.go

```diff
@@ -921,11 +900,10 @@ func (j *ResizeJob) setState(state string) {
 
 // Run distributes ResizeInstructions.
 func (j *ResizeJob) Run() error {
-	j.mu.RLock()
-	defer j.mu.RUnlock()
-
 	// Set job state to RUNNING.
+	j.mu.RLock()
 	j.setState(ResizeJobStateRunning)
+	j.mu.RUnlock()
```

Comment:
looks like that would need to be a write lock

Reply:
good catch. i pushed a fix.